test/ConnectionTest.cpp

```diff
@@ -25,4 +35,14 @@ namespace {
         ASSERT_EQ("5.5.8 MySQL Community Server (GPL)", getConnectionAdvancedInfo(connection));
     }
 
+    TEST(ConnectionInfoTest, GetDInfo) {
+        MySqlDBConnection connection("Some server", 7);
+        ASSERT_EQ(7, getDConnectionInfo(connection));
+    }
+
+    TEST(ConnectionInfoTest, GetDAdvancedInfo) {
```

Comment:
@Morheit , add test which will fall back to dynamic cast.
also take into consideration my comments from prev PR regarding putting everything to the same sources and naming. Also your test names are not meaningful.  Consier "GetDAdvancedInfo" -> "GetAdvacedType3StaticCast". Imagine you have console output with test run results were all test cases looks the same. it's hard to understand where  to find test and what might went wrong quickly.

Reply:
@malirod , done